lib/airflow/airflow/contrib/jobs/event_based_scheduler_job.py

```diff
@@ -270,11 +270,12 @@ def _register_periodic_events(self, execution_date, dag, session=None):
     @provide_session
     def _remove_periodic_events(self, dag_id, execution_date, session=None):
         dagruns = DagRun.find(dag_id=dag_id, execution_date=execution_date)
-        dag = self.dagbag.get_dag(dag_id=dagruns[0].dag_id, session=session)
-        for task in dag.tasks:
-            if task.executor_config is not None and 'periodic_config' in task.executor_config:
-                self.log.debug('remove periodic task {} {} {}'.format(dag_id, execution_date, task.task_id))
-                self.periodic_manager.remove_task(dag_id, execution_date, task.task_id)
+        if len(dagruns) > 0:
```

Comment:
Hi @aqua7regia , thanks for your pull request.
IMO, it deserves a WARNING log if `len(dag_run) = 0` which helps debug, and please add a unit test for the change.